### PR TITLE
(#3357) - Validate attachment names dont start with _

### PR DESCRIPTION
**Pull request by daleharvey**

### CODE REVIEW

**Comment by nolanlawson** [comment]
+1 when green.

File: lib/adapter.js [hunk 1]
```diff
@@ -153,6 +153,13 @@ function doNextCompaction(self) {
   });
 }
 
+function attachmentNameError(name) {
+  if (name.charAt(0) === '_') {
+    return 'Attachment name can\'t start with \'_\'';
```
Comment (kxepal) [inline lib/adapter.js]: Why not to include bad attachment name into error? This will save time to figure out what exactly caused the issue.

Comment (daleharvey) [inline lib/adapter.js]: Yeh will probably will do that on the way in, I was doing compatibility with Couch but them we avoid testing for exact error message so itll be fine

Comment (kxepal) [inline lib/adapter.js]: Let's improve Couch errors as well! (:

**Comment by daleharvey** [comment]
Cheers, merged with a more descriptive error - https://github.com/pouchdb/pouchdb/commit/57bc143a3920ecf1925ffa346d2d83e80520683d

**Pull request closed without merging**